cmd/ferretdb/readyz.go

```diff
@@ -75,27 +76,46 @@ func (r *ReadyZ) Probe(ctx context.Context) bool {
 	for _, u := range urls {
 		r.l.DebugContext(ctx, fmt.Sprintf("Pinging %s", u))
 
-		client, err := mongo.Connect(ctx, options.Client().ApplyURI(u))
+		conn, err := wireclient.Connect(ctx, u, r.l)
 		if err != nil {
 			r.l.ErrorContext(ctx, "Connection failed", logging.Error(err))
 			return false
 		}
 
-		pingErr := client.Ping(ctx, nil)
+		defer func() {
+			if err = conn.Close(); err != nil {
+				r.l.ErrorContext(ctx, "Closing connection failed", logging.Error(err))
+			}
+		}()
 
-		err = client.Disconnect(ctx)
-
-		if pingErr != nil {
-			r.l.ErrorContext(ctx, "Ping failed", logging.Error(pingErr))
+		_, resBody, err := conn.Request(ctx, wire.MustOpMsg(
+			"ping", int32(1),
+			"$db", "admin",
+		))
+		if err != nil {
+			r.l.ErrorContext(ctx, "Ping request failed", logging.Error(err))
 			return false
 		}
 
+		res, err := must.NotFail(resBody.(*wire.OpMsg).RawDocument()).Decode()
 		if err != nil {
-			r.l.ErrorContext(ctx, "Disconnect failed", logging.Error(err))
+			r.l.ErrorContext(ctx, "Decoding ping response failed", logging.Error(err))
```

Comment:
In wireclient, what do you think of adding a helper function that returns a document for successful response with ok=1 and go error for ok=0?

Reply:
How would you call and what documentation comment would say?

Reply:
Below is something I'm thinking about. I think it's nice to be able to use logging by `r.l.ErrorContext(ctx, "Ping failed", logging.Error(err))` instead of decoding each field like `code` `codeName` and `errmsg`.
```
// RequestMsg sends the op message request to the connection and
// returns decoded response.
// It returns CommandError if the response returned MongoDB compatible error.
(c *Conn) RequestMsg(ctx context.Context, body *wire.OpMsg) (*wirebson.Document, error) {
...
}

type CommandError struct{
    Code int
    Name string
    Message string
}

func (err *CommandError) Error() string{
...
}
```